metricflow-semantics/metricflow_semantics/specs/patterns/minimum_time_grain.py

```diff
@@ -48,20 +47,23 @@ class MinimumTimeGrainPattern(SpecPattern):
     def match(self, candidate_specs: Sequence[InstanceSpec]) -> Sequence[InstanceSpec]:
         spec_set = group_specs_by_type(candidate_specs)
 
-        spec_key_to_grains: Dict[TimeDimensionSpecComparisonKey, Set[TimeGranularity]] = defaultdict(set)
+        spec_key_to_grains: Dict[TimeDimensionSpecComparisonKey, Set[ExpandedTimeGranularity]] = defaultdict(set)
         spec_key_to_specs: Dict[TimeDimensionSpecComparisonKey, List[TimeDimensionSpec]] = defaultdict(list)
         for time_dimension_spec in spec_set.time_dimension_specs:
             spec_key = time_dimension_spec.comparison_key(exclude_fields=(TimeDimensionSpecField.TIME_GRANULARITY,))
-            spec_key_to_grains[spec_key].add(time_dimension_spec.time_granularity.base_granularity)
+            spec_key_to_grains[spec_key].add(time_dimension_spec.time_granularity)
             spec_key_to_specs[spec_key].append(time_dimension_spec)
 
         matched_time_dimension_specs: List[TimeDimensionSpec] = []
         for spec_key, time_grains in spec_key_to_grains.items():
-            matched_time_dimension_specs.append(
-                spec_key_to_specs[spec_key][0].with_grain(
-                    ExpandedTimeGranularity.from_time_granularity(min(time_grains))
-                )
+            sorted_time_grains = sorted(
+                time_grains,
+                # Sort by smallest to largest standard granularity, with custom granularities last (sorted by their
+                # base granularity) since we don't know how large they are.
+                key=lambda grain: (grain.is_custom_granularity, grain.base_granularity.to_int()),
```

Comment:
Idea: Would it be nicer if we added the comparison methods to `ExpandedTimeGranularity` so you can just sort it as normal
```python
def __lt__(self, other):
        return self._cmp(other) < 0

    def __le__(self, other):
        return self._cmp(other) <= 0

    def __eq__(self, other):
        return self._cmp(other) == 0

    def __ne__(self, other):
        return self._cmp(other) != 0

    def __ge__(self, other):
        return self._cmp(other) >= 0

    def __gt__(self, other):
        return self._cmp(other) > 0
```
might be more forward thinking tho since i don't think it's used in multiple places

Reply:
That's a good idea 🙂‍↕️

Reply:
Actually, I changed my mind. Only because we can't actually compare the size of custom granularities - we have no idea which ones are bigger than the others. So we can sort by base granularity but then after that it would be random. Here it doesn't matter in practice, so this sorting method is fine, but we probably shouldn't add a partly-random sorting method that might be used elsewhere without knowing the context.